### DISCUSSION TITLE
How to interpret the pull shark badge?

### DISCUSSION THREAD

**Discussion by sauntimo** [community/community#18248] [Other Feature Feedback, Questions, & Ideas] [answered]
Love the new achievement badges, and think they are a really good idea. But, I was confused by this:

![image](https://user-images.githubusercontent.com/2720466/173039137-64b3dc01-3875-473b-8a40-2a1d810d5a75.png)

According to the screenshot I have earned this badge 3 times, does that mean 3 pull requests that have been merged? However, it says "inaccessible" next to 2nd pull request merged - is that a bug? or does it think I haven't had a second pull request merged? In the last 6 years I've had several hundred merged for various companies I've worked for so does it only count merged PRs in public repositories? It doesn't say that.

What does bronze and silver mean? were they both unlocked on 28th Nov 2018? How does the "x3" relate to the "bronze and silver" or the "2nd, 16th, 128th" it mentions variously?

I don't really understand how to interpret any of this, and whilst I might just be being a bit slow today, I've used github for years so it makes me think new users might also be confused.

**Comment by DavidYang2149** [comment, marked as answer]
Hi~!
Pull Shark seem to be check your PR record and show Level (not your PR count status)
x1 - over 2 PR
x2 - over 16 PR
x3 - over 128 PR
x4 - over 1024 PR

Also "inaccessible" seem to be private PR.
I agree new users might be confused.

**Reply by sauntimo** [reply]
so does that mean
- when over 2 PRs are merged, you reach level 1 (x1) and you earn the initial badge
- when over 16 PRs are merged, you reach level 2 (x2) and you earn the bronze badge
- when over 128 PRs are merged, you reach level 3 (x3) and you earn the silver badge
- when over 1024 PRs are merged, you reach level 4 (x4) and you earn the gold badge

because if that's what they mean, they really haven't communicated that effectively.

**Reply by johnedquinn** [reply]
I was going to ask the same thing. I agree. Not communicated effectively.

IMO it should be L1 to denote Level 1. L1, L2, L3, and so on. And when you click on the bage, it should show all 4 levels and where the user is at. Especially should show how many PRs they've merged overall (to see how close they are to the next level).

**Reply by odil-io** [reply]
Depends on how you look at it I guess, I mean, I could say that I've earned that badge 3 times instead _with_ 3 merged PR's. I do get the confusion tho as I was confused the first few times too.
But one day it just clicked with "I earned this X, Y times".

But I'd rather see a different badge than how many times I've earned a particular one.

**Reply by vhirtham** [reply]
I agree with @sauntimo and @johnedquinn that the current presentation of this badge is somewhat unintuitive and easy to misunderstand. I made some suggestions in [this related discussion](https://github.com/orgs/github-community/discussions/18298) that might help avoiding further confusion.

**Reply by ghost** [reply]
> so does that mean
> 
> * when over 2 PRs are merged, you reach level 1 (x1) and you earn the initial badge
> * when over 16 PRs are merged, you reach level 2 (x2) and you earn the bronze badge
> * when over 128 PRs are merged, you reach level 3 (x3) and you earn the silver badge
> * when over 1024 PRs are merged, you reach level 4 (x4) and you earn the gold badge
> 
> because if that's what they mean, they really haven't communicated that effectively.

I have earnt before (check my profile), and it is actually the case. "inaccessible" means "private repository".

**Reply by gustavofreze** [reply]
> I was going to ask the same thing. I agree. Not communicated effectively.
> 
> IMO it should be L1 to denote Level 1. L1, L2, L3, and so on. And when you click on the bage, it should show all 4 levels and where the user is at. Especially should show how many PRs they've merged overall (to see how close they are to the next level).

I agree that there should be this feature of showing progress.
It's not the ideal way, but you can track where you are by looking at your closed [pulls](https://github.com/pulls).

**Reply by LukeLIN-web** [reply]
Yeah, `x2` is confusing, I think it should be Level 2.

**Reply by beingshahidali** [reply]
Hey all!  Just got the Pull Shark 3 badge (128+ merged PRs)! 
I agree the badge system could use better clarity—seeing progress would be motivating. 
My tip: focus on consistent PRs in both public and private repos to level up faster. Let’s connect and share more coding tips! 

 Follow for more!"💪

**Reply by R-Tharanka** [reply]
Currently, I have 168 closed public pull requests, but my profile's achievements section shows x2. 
![image](https://github.com/user-attachments/assets/0a7e4175-593f-4fa5-99c0-1e045f6c9239)
What could be the reason for this?

**Comment by devcrafta** [comment]
We can find out more about Badge here.
https://github.com/Schweinepriester/github-profile-achievements

**Reply by mribeirodantas** [reply]
Thanks for sharing, @Astro2020-lovely !

**Reply by regidf12** [reply]
Thanks for the info

**Reply by devcrafta** [reply]
You're welcome!!! :)

**Reply by Gyshie** [reply]
I was looking for this. Thanks!

**Reply by Monoget** [reply]
Thanks

**Reply by parkgeonwoong** [reply]
Thanks for the info

**Reply by mustafaihsank** [reply]
Thanks :)

**Reply by messiisgreat** [reply]
Thanks!

**Reply by Hammad-Hasnain** [reply]
I've been looking for this, Thanks for sharing...

**Comment by gabrielporfiro** [comment]
Após responder duas discussões você obtém o galaxy brain

**Reply by mribeirodantas** [reply]
Interessante. Será que todos os achievements seguirão essa mesma regra?

**Reply by gabrielporfiro** [reply]
> Interessante. Será que todos os achievements seguirão essa mesma regra?

cada um tem seu motivo, tem no repositorio que ele mostrou ai:
https://github.com/Schweinepriester/github-profile-achievements

**Comment by breakdowns** [comment]
Cool

**Reply by Monoget** [reply]
Great

**Comment by YouthguyZ** [comment]
➡️[https://github.com/Schweinepriester/github-profile-achievements](url)

**Comment by jsoref** [comment]
The inaccessible bit should mean that you no longer have access to the repository where the PR was merged. If you worked for orgs and no longer do, that'd explain why it'd be easy for your PRs to land that way.

I do wonder what happens if your PR#2 was merged into a repository that was then deleted. Does that PR no longer count? My naive guess is that it wouldn't count, but anything is possible. (I know someone who has made 0 PRs, so we could actually test this.)

**Comment by vinkay215** [comment]
Any news about this?

**Reply by saitiger** [reply]
Inaccessible means that the pull requests and merges were either made in a private repo or the repo has been deleted since. The other thing is the number of pull requests that were merged. There are three 4 tiers. Default - 2, Bronze - 16, Silver - 128 and Gold - 1024. 
You can learn more about badges [here](https://github.com/Thinkright20/Profile-Badges).

**Reply by messiisgreat** [reply]
Thanks!

**Comment by roguh** [comment]
When will you add the next tier? >1024

**Comment by abhisek2004** [comment]
<img width="435" height="260" alt="image" src="https://github.com/user-attachments/assets/0f2c229a-6436-4b0f-91a2-85b58d90feed" />


Hey all! Just got the Pull Shark 3 badge (128+ merged PRs)!
I agree the badge system could use better clarity—seeing progress would be motivating.
My tip: focus on consistent PRs in both public and private repos to level up faster. Let’s connect and share more coding tips!

Follow for more!"💪
